### PR TITLE
state: add minimal taskrunner logging

### PR DESCRIPTION
**Pull request by mvo5**
And also avoid logging all http handling of the /operations/ calls that happens 10-20 times per sec.

### CODE REVIEW

**Comment by zyga** [comment]
+1

It would be great to log all task.Logf, task.Errorf and handler error results as well

**Comment by mvo5** [comment]
@zyga thanks! Happy to add more logging, i.e.:
```
diff --git a/overlord/state/task.go b/overlord/state/task.go
index 91ce523..2d98583 100644
--- a/overlord/state/task.go
+++ b/overlord/state/task.go
@@ -22,6 +22,8 @@ package state
 import (
        "encoding/json"
        "fmt"
+
+       "github.com/ubuntu-core/snappy/logger"
 )
 
 type progress struct {
@@ -199,7 +201,9 @@ func (t *Task) addLog(kind, format string, args []interface{}) {
                copy(t.log, t.log[len(t.log)-9:])
                t.log = t.log[:9]
        }
-       t.log = append(t.log, fmt.Sprintf(kind+": "+format, args...))
+       msg := fmt.Sprintf(kind+": "+format, args...)
+       t.log = append(t.log, msg)
+       logger.Debugf(msg)
 }
 
 // Log returns the most recent messages logged into the task.
```
is really simple to add if there is consensus on this.

File: overlord/state/taskrunner.go [hunk 1]
```diff
@@ -229,6 +231,7 @@ func (r *TaskRunner) Ensure() {
 			// Dependencies still unhandled.
 			continue
 		}
+		logger.Debugf("running task id %q: %q", t.ID(), t.Summary())
```
Comment (niemeyer) [inline overlord/state/taskrunner.go]: `Running task %s: %s`?

Log messages are generally capitalized, I think.

File: daemon/daemon.go [hunk 2]
```diff
@@ -145,7 +146,9 @@ func logit(handler http.Handler) http.Handler {
 		t0 := time.Now()
 		handler.ServeHTTP(ww, r)
 		t := time.Now().Sub(t0)
-		logger.Debugf("%s %s %s %s %d", r.RemoteAddr, r.Method, r.URL, t, ww.s)
+		if !strings.Contains(r.URL.String(), "/operations") {
```
Comment (niemeyer) [inline daemon/daemon.go]: Needs to sync with #946. Whoever lands second will need to be adapted.

Comment (niemeyer) [inline daemon/daemon.go]: Perhaps `|| Contains(..., "/changes"), so it remains valid?

**Comment by niemeyer** [comment]
That sounds good too @mvo5. I'd just recommend saving the Sprintf output so we're not doing it twice.

**Comment by mvo5** [comment]
Thanks! addressed the feedback and added some suggestions from telegram. thanks @chipaca for this.

**Comment by mvo5** [comment]
retest this please

**Comment by niemeyer** [comment]
retest this please

**Comment by mvo5** [comment]
retest this please

**Pull request merged**